src/main/java/fi/aalto/cs/apluscourses/model/News.java

```diff
@@ -60,14 +59,16 @@ public static News fromJsonObject(@NotNull JSONObject object, @NotNull Course co
     var body = object.getString("body");
     var bodyElement = Jsoup.parseBodyFragment(body).body();
 
-    NewsParser parser;
-    switch (course.getName()) {
-      case "O1":
-        parser = new O1NewsParser(language);
-        break;
-      default:
-        parser = new DefaultNewsParser();
-        break;
+    var parser = new NewsParser();
```

Comment:
`parser` shouldn't be initialized with a new object instance which may be immediately discarded.